fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -108,6 +109,9 @@ public LucenePartitioner(@Nonnull IndexMaintainerState state) {
         if (partitioningEnabled && (partitionTimestampFieldName.isEmpty() || partitionTimestampFieldName.isBlank())) {
             throw new RecordCoreArgumentException("Invalid partition timestamp field name", LogMessageKeys.FIELD_NAME, partitionTimestampFieldName);
         }
+        // partition timestamp field name in lucene, when nested, has `_` in place of `.`
+        partitionTimestampFieldNameInLucene = partitionTimestampFieldName == null ? null : partitionTimestampFieldName.replace('.', '_');
+
```

Comment:
I wonder whether this could be done at a higher level so that here we only see a consistent naming scheme. Not needed for this PR, though.